docs/en/observability/monitor-kubernetes.asciidoc

```diff
@@ -0,0 +1,223 @@
+Kubernetes, host, and the cloud providers.
+
+* (optional) Drop unwanted events
+
+* Deploy Filebeat as a DaemonSet on Kubernetes
+
+[discrete]
+=== View logs in {kib}
+
+TODO: Describe how to view logs in Kibana. Show how to use the Logs app and how to
+view pre-built dashboards and visualizations.
+
+[discrete]
+== Part 2: Monitor health and performance metrics
+
+TODO: provide a brief intro. We'll need to figure out how much to cover here vs
+the monitoring overview. Again lots of good introductory info under
+"K8s metrics collection with Metricbeat"
+https://www.elastic.co/blog/kubernetes-observability-tutorial-k8s-metrics-collection-and-analysis[here].
+
+
+[discrete]
+=== Metrics you should monitor
+
+TODO: Discuss the metrics that users should monitor: cluster-level
+metrics from Kubernetes, pod metrics, metrics about nodes, technology-specific
+metrics (like nginx),etc. Explain where the metrics come from (kubelet API,kube-state-metrics,
+apiserver, etc) and why to monitor them.
+
+
+[discrete]
+=== Deploy {metricbeat} to collect metrics
+
+TODO: Describe how to configure and deploy Metricbeat as a daemonset. Rough list
+of what we need to cover:
+
+* Download the Metricbeat deployment manifest
+
+* Set the connection information for Elasticsearch (also describe how to create
+secrets or point to earlier section for reminder)
+
+* Configure metrics collection:
+
+** Collect system metrics.
+
+** Collect Docker metrics.
+
+** Collect Kubernetes pod metrics.
+
+** Collect Kubernetes state metrics
+
+** Collect application-specific metrics (use hint-based autodiscovery).
+Examples: NGINX, MySQL
+
+** Collect Prometheus metrics.
+
+* Add metadata to events. Describe how the events are enriched with
+metadata coming from Docker, Kubernetes, host, and the cloud providers
+
+* Deploy Metricbeat as a DaemonSet on Kubernetes
+
+[discrete]
+=== View performance and health metrics
+
+TODO: Describe how to use the Metrics app and pre-built dashboards/visualizations.
+
+[discrete]
+== Part 3: Monitor uptime and availability data
+
+TODO: provide a brief intro.
+
+[discrete]
+=== Deploy {heartbeat} to collect uptime and availability data
+
+TODO: Describe how to configure and deploy Heartbeat as a daemonset.
+
+[discrete]
+=== View uptime and availability in {kib}
+
+TODO: Describe how to use the Uptime app and pre-built dashboards/visualizations.
+
+[discrete]
+== Part 4: Monitor internal network traffic data
+
+TODO: provide a brief intro.
+
+[discrete]
+=== Deploy {packetbeat} to capture network traffic data
+
+TODO: Describe how to configure and deploy Packetbeat as a daemonset.
+
+[discrete]
+=== View network traffic data
+
+TODO: Describe how to use the pre-built dashboards/visualizations.
+
+[discrete]
+== Part 5: Monitor application performance
+
+TODO: Describe how to use APM. This part needs more sections. I'm leaving the
+sections out here because I'm not sure what we should include.
```

Comment:
A proposed section breakdown:
- Setup
  - Setup APM Server
    - Through [ECK](https://www.elastic.co/guide/en/cloud-on-k8s/1.2/k8s-apm-server.html)
    - On cloud
    - [Download and install](https://www.elastic.co/guide/en/apm/server/current/installing.html)
  - Setup APM Agents
    - Java agent (see https://www.elastic.co/blog/using-elastic-apm-java-agent-on-kubernetes-k8s)
    - NodeJS Agent
    - Python Agent
    - ... and so forth
- Configure
  - eg https://www.elastic.co/guide/en/apm/server/7.9/metadata-api.html#kubernetes-data

Reply:
Will add to the outline. thanks!